src/gui/properties/proplistdelegate.h

```diff
@@ -48,25 +49,27 @@ enum PropColumn
     AVAILABILITY
 };
 
-class PropListDelegate final : public ProgressBarDelegate
+class PropListDelegate final : public QStyledItemDelegate
 {
     Q_OBJECT
+    Q_DISABLE_COPY(PropListDelegate)
 
 public:
     explicit PropListDelegate(PropertiesWidget *properties);
 
-    void setEditorData(QWidget *editor, const QModelIndex &index) const override;
-    QWidget *createEditor(QWidget *parent, const QStyleOptionViewItem & /* option */, const QModelIndex &index) const override;
-
 public slots:
     void setModelData(QWidget *editor, QAbstractItemModel *model, const QModelIndex &index) const override;
-    void updateEditorGeometry(QWidget *editor, const QStyleOptionViewItem &option, const QModelIndex & /* index */) const override;
+    void updateEditorGeometry(QWidget *editor, const QStyleOptionViewItem &option, const QModelIndex &index) const override;
 
 signals:
     void filteredFilesChanged() const;
 
 private:
-    void initProgressStyleOption(QStyleOptionProgressBar &option, const QModelIndex &index) const override;
+    void setEditorData(QWidget *editor, const QModelIndex &index) const override;
```

Comment:
Why do you redeclare these methods as `private`?

Reply:
should they not be private?

Reply:
They're previously declared as `public`, so No.

Reply:
>They're previously declared as public, so No.

Why they need to be public?

Reply:
They're parts of public interface of QAbstractItemDelegate. Since they're virtual it's at least meaningless to hide them.

Reply:
>Why they need to be public?

This is the wrong question. I mentioned recently that it doesn't make sense to intentionally hide something that can be used correctly from the outside. Only the actual implementation details should be private.

Reply:
fixed.